sktime/utils/validation/series.py

```diff
@@ -153,3 +158,45 @@ def check_equal_time_index(*ys):
 
         if not first_index.equals(y.index):
             raise ValueError("Some (time) indices are not the same.")
+
+
+def _is_int_index(index):
+    """Check if index type is one of pd.RangeIndex or pd.Int64Index"""
+    return type(index) in VALID_INDEX_TYPES[:2]
+
+
+def check_consistent_index_type(*ys):
+    """Check that index types are compatible for plotting.
+
+    Parameters
+    ----------
+    ys : pd.Series
+        One or more time series
+
+    Returns
+    -------
+    index : pd.Index
+        Validated and combined indexes
+
+    Raises
+    ------
+    TypeError
+        If index types are inconsistent
+    """
+
+    index = ys[0].index
+    for y in ys[1:]:
+        # check types, note that isinstance() does not work here because index
+        # types inherit from each other, hence we check for type equality
+        msg = "Please make sure that all series have consistent index type."
+
+        if _is_int_index(y.index):
+            if not _is_int_index(index):
+                raise TypeError(msg)
+
+        else:
+            if not type(index) is type(y.index):  # noqa
+                raise TypeError(msg)
+
+        index = index.union(y.index)
```

Comment:
I wouldn't include this operation into this function, it's not directly related to the checking of consistent index types.